4/debian-10/rootfs/opt/bitnami/scripts/libpgpool.sh

```diff
@@ -524,15 +540,28 @@ pgpool_generate_password_file() {
     if is_boolean_yes "$PGPOOL_ENABLE_POOL_PASSWD"; then
         info "Generating password file for local authentication..."
 
-        pg_md5 -m --config-file="$PGPOOL_CONF_FILE" -u "$PGPOOL_POSTGRES_USERNAME" "$PGPOOL_POSTGRES_PASSWORD"
+        local -a password_encryption_cmd=("pg_md5")
+
+        if [[ "$PGPOOL_AUTHENTICATION_METHOD" = "scram-sha-256" ]]; then
+
+            if is_file_writable "$PGPOOLKEYFILE"; then 
+                # Creating a PGPOOLKEYFILE as it is writeable
+                echo "$PGPOOL_AES_KEY" > "$PGPOOLKEYFILE"
+            fi
+            # Fix permissions for PGPOOLKEYFILE
+            chmod 0600 "$PGPOOLKEYFILE"
```

Comment:
We should also configure it so that permissions are not changed if the file is not writable, or it could cause issues when mounting it.
```suggestion
                # Fix permissions for PGPOOLKEYFILE
                chmod 0600 "$PGPOOLKEYFILE"
            fi
```

Reply:
Got it. Thanks.